src/ngResource/resource.js

```diff
@@ -721,7 +718,13 @@ angular.module('ngResource', ['ng']).
                 (success || noop)(value, response.headers);
                 return value;
               },
-              responseErrorInterceptor);
+              responseErrorInterceptor || error ?
+              function(response) {
+                (error || noop)(response);
+                (responseErrorInterceptor || noop)(response);
+                return response;
```

Comment:
The proposed alternative has the issue that if `responseErrorInterceptor` is undefined, then `return $q.reject(response);` will be unhandled.